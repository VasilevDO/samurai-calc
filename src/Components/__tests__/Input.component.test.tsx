import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import Input from '../Input.component';

describe('InputComponent', () => {
	const testId = 'input';
	const initValue = 'value';
	const changeAction = jest.fn();

	test('should render a correct element', () => {
		const {container, getByTestId} = render(<Input value={initValue} onChange={changeAction}/>);
		const element = getByTestId(testId);
		expect(element).toBeInTheDocument();
		expect(element).toHaveValue(initValue);
		expect(container).toMatchSnapshot();
	});
});

