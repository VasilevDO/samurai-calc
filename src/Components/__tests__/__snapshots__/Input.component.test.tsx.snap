// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent should render a correct element 1`] = `
.c0 {
  width: 100%;
  height: 76px;
  color: inherit;
  text-align: right;
  background-color: inherit;
  border: none;
  margin-bottom: 18px;
  font-weight: 700;
  font-size: 56px;
  line-height: 80px;
}

.c0:focus {
  outline: none;
}

@media only screen and (max-width:768px) {
  .c0 {
    height: 1.5em;
    font-weight: 500;
    font-size: 28px;
    line-height: 40px;
  }

  .c0 >:not(:last-child) {
    margin-bottom: 14px;
  }
}

<div>
  <input
    class="c0"
    data-testid="input"
    value="value"
  />
</div>
`;
